packages/x-components/src/components/__tests__/snippet-callbacks.spec.ts

```diff
@@ -0,0 +1,94 @@
+import { mount, Wrapper } from '@vue/test-utils';
+import { installNewXPlugin } from '../../__tests__/utils';
+import { baseSnippetConfig } from '../../views/base-config';
+import { XEventListeners } from '../../x-installer/api/api.types';
+import SnippetCallbacks from '../snippet-callbacks.vue';
+
+function renderSnippetCallbacks({
+  callbacks = {}
+}: RenderSnippetCallbacksOptions = {}): RenderSnippetCallbacksAPI {
+  const [, localVue] = installNewXPlugin();
+  const wrapper = mount(
+    {
+      components: { SnippetCallbacks },
+      template: '<SnippetCallbacks />'
+    },
+    {
+      provide: {
+        snippetConfig: localVue.observable({ ...baseSnippetConfig, callbacks })
+      },
+      localVue
+    }
+  );
+
+  return {
+    wrapper
+  };
+}
+
+describe('testing SnippetCallbacks component', () => {
+  it('executes a callback injected from the snippetConfig', () => {
+    const acceptedAQueryCallback = jest.fn(payload => payload);
+    const clickedColumnPickerCallback = jest.fn(payload => payload);
+    const { wrapper } = renderSnippetCallbacks({
+      callbacks: {
+        UserAcceptedAQuery: acceptedAQueryCallback,
+        UserClickedColumnPicker: clickedColumnPickerCallback
+      }
+    });
+
+    wrapper.vm.$x.emit('UserAcceptedAQuery', 'lego');
+
+    expect(acceptedAQueryCallback).toHaveBeenCalledTimes(1);
+    expect(acceptedAQueryCallback).toHaveBeenCalledWith('lego', {
+      location: undefined,
+      moduleName: null
+    });
+
+    expect(clickedColumnPickerCallback).not.toHaveBeenCalled();
```

Comment:
After this emit the `UserClickedColumnPicker` and do `toHaveBeenCalledTimes` assertions for both events and `toHaveBeenCalledWith` for the column one to check that all events passed as callbacks would work as expected.

Reply:
I don't know if I follow you. In this case we assert the function not no be called because the event is not emitted. This way we are testing that when an event is being emitted only the correct callback is triggered, and not the rest of callbacks.

Reply:
Here you test that the callback works for one event and doesn't affect the others. I'm suggesting that you emit the other event too to check that the callbacks work for any* event and the callbacks are called only when the event they're linked to is called. Maybe it's too much so adding more is your call.

This is what I had in mind:

```suggestion
    expect(clickedColumnPickerCallback).not.toHaveBeenCalled();
    
    wrapper.vm.$x.emit('UserClickedColumnPicker', 1);

    expect(acceptedAQueryCallback).toHaveBeenCalledTimes(1);
    expect(clickedColumnPickerCallback).toHaveBeenCalledTimes(1);
```

Reply:
Done!